### ISSUE TITLE
Can't get both a file upload and a text input

### ISSUE THREAD

**Issue by pipoprods** [ColasNahaboo/cgibashopts#7]
Hi,

I'm trying to convert a bash script to a CGI. This script takes a file and an text field as inputs.

My form looks like this:

```html
<form method="POST" action="/backend.sh" enctype="multipart/form-data">
  <input type="file" name="file" />                                    
  <input type="text" name="text" />                                    
  <button type="submit">Go</button>                                    
</form>                                                                
```

On the backend I receive the file properly:

```
FORMFILE_file=/tmp/cgibashopts-files.8/file
FORMS=file
```

But I don't receive the text value.

If I remove the file input :

```html
<form method="POST" action="/backend.sh" enctype="multipart/form-data">
  <input type="text" name="text" />                                    
  <button type="submit">Go</button>                                    
</form>                                                                
```

Then I get the text value:

```
FORMS=text
FORM_text=dqdfsq
FORMFILES=
FORMQUERY=
```

Is it possible to handle both? Am I using it a wrong way?

**Comment by ColasNahaboo** [comment]
You are using it properly.
It seems you just **found a bug** in cgibashopts. It seems to ignore fields present **after** a `file` field.

E.g, this works:

```
<form method="POST" action="cgi-bin/test-cgbo.cgi" enctype="multipart/form-data">
  <input type="text" name="text" value="foo"/>                                    
  <input type="file" name="file" />                                    
  <button type="submit">Go</button>                                    
</form>                                                                
==> FORMS="text file"
```
But not:

```
<form method="POST" action="cgi-bin/test-cgbo.cgi" enctype="multipart/form-data">
  <input type="file" name="file" />                                    
  <input type="text" name="text" value="foo"/>                                    
  <button type="submit">Go</button>                                    
</form> 
==> FORMS="file"                                                               
```
And it ignores any type of field, not just text.

I will look on how to fix it. It should not be too long (a day or two?), but in the meantime you can use the workaround of making sure the file field is the last on in the form.

Thanks a lot for reporting the bug, and taking the time to provide a code sample!

**Comment by pipoprods** [comment]
I think you forgot to swap lines in your code examples :wink: 

Thanks for the workaround :slightly_smiling_face:

**Comment by ColasNahaboo** [comment]
Oops, yes, corrected!

**Comment by ColasNahaboo** [comment]
Ok, fixed!
I published the v4.1.1 release with the fix. https://github.com/ColasNahaboo/cgibashopts/releases/tag/v4.1.1

**Comment by pipoprods** [comment]
Great! Thanks a lot :slightly_smiling_face:
